src/main/java/frc/robot/subsystems/shooter/Shooter.java

```diff
@@ -41,16 +41,22 @@ public void setBottomSpeed(double speed) {
     bottomMotor.setSpeed(speed);
   }
 
-  Debouncer speedDebouncer = new Debouncer(0.4);
-  public boolean atSpeed() {
+  Debouncer speedDebouncer = new Debouncer(0.2);
+  public boolean atSpeed(double overrideTopSetpoint, double overrideBottomSetpoint) {
     double tolerance = SPEED_TOLERANCE;
+    overrideTopSetpoint = Math.signum(topMotor.getVelocitySetpoint()) * Math.abs(overrideTopSetpoint);
```

Comment:
move the formulas here into their own method,ex. 
calculateOverrideSetpoint(val1, val2) { return Math.signum(val1)*Math.abs(val2); }

Reply:
it's not really a formula, it's just matching the sign of overrideTopSetpoint to what is on the current setpoint. i'd agree if it was any more complex though

Reply:
I would still convert it into it's own method, even if small - makes the code easier to read and it is reusable